### PR TITLE
It is now possible to login using either email or nickname

### PR DESCRIPTION
**Pull request by ggc87**

### CODE REVIEW

File: emotly/controllers/user_controller.py [hunk 1]
```diff
@@ -27,7 +27,10 @@ def login():
     try:
         # Retrieve json data and user data.
         data = json.loads(request.data.decode('utf-8'))
-        user = User.objects.get(email=data['email'])
+        if 'email' in data:
+            user = User.objects.get(email=data['email'])
+        else:
+            user = User.objects.get(nickname=data['nickname'])
```
Comment (tizianasellitto) [inline emotly/controllers/user_controller.py]: We should add both of the fields? Hmmmmm Why not only a "login" field and check in the controller if it's an email (just with the presence of the @) or a nickname?

Comment (ggc87) [inline emotly/controllers/user_controller.py]: You are right I even thought about that and than I don't know why I did that.  I had  SiGSEGV.  Anyway I don't know if the presence of @ can work, can a nickname contain that character?

File: emotly/controllers/user_controller.py [hunk 2]
```diff
@@ -27,7 +27,10 @@ def login():
     try:
         # Retrieve json data and user data.
         data = json.loads(request.data.decode('utf-8'))
-        user = User.objects.get(email=data['email'])
+        if '@' in data['user_id']:
+            user = User.objects.get(email=data['user_id'])
+        else:
+            user = User.objects.get(nickname=data['user_id'])
```
Comment (michelangelo) [inline emotly/controllers/user_controller.py]: Curiosity: are we sure `user` is gonna be visible in the outer scope??

**Comment by michelangelo** [comment]
Almost LGTM.

Please:
* add coverage for username login cases as they are completely missing
* review the *curiosity* question in the `user_controller`.
* check why the CI is failing here.

Also, **DO NOT MERGE** THIS until we add a second commit to change the login requests in the PWA.

@tizianasellitto, if @ggc87 is not available today would you please check this as it's blocking the rest? Thanks.

**Comment by michelangelo** [comment]
@ggc87 @tizianasellitto: rebased and added change to PWA.

**Comment by tizianasellitto** [comment]
- I think there is enough test coverage for the login feature. 
- Python does not have separate block-level scope for try and except, AFAIK. So user is visible in the whole function.
- The CI keeps failing due to something related to the verify token (I suppose)

**Comment by michelangelo** [comment]
Alright, green light for me, then.

**Pull request merged**